src/main/kotlin/at/yawk/katbot/action/NCov.kt

```diff
@@ -31,117 +31,8 @@ class NCov @Inject constructor(private val eventBus: EventBus) {
     @Subscribe
     fun command(command: Command) {
         if (command.line.messageIs("ncov")) {
-            if (!command.public) {
-                command.channel.sendMessageSafe("Not here, sorry.")
-                throw CancelEvent
-            }
-
-            val cached = this.cache
-            val now = Instant.now()
-            var result: List<Region>
-            if (cached == null || cached.time < now.minusSeconds(TimeUnit.MINUTES.toSeconds(15))) {
-                result = try {
-                    load()
-                } catch (e: Exception) {
-                    command.channel.sendMessageSafe(e.toString())
-                    throw CancelEvent
-                }
-                this.cache = DataWithTime(now, result)
-            } else {
-                result = cached.result
-            }
-
-            result = result.sortedBy { -it.cases }
-
-            val total = Region(
-                    "Total",
-                    cases = result.sumBy { it.cases },
-                    deaths = result.sumBy { it.deaths },
-                    recoveries = result.sumBy { it.recoveries })
-            result = listOf(total) + result
-
-            val messageBuilder = StringBuilder((command.target ?: command.actor).nick).append(", nCov cases: ")
-
-            for ((i, region) in result.withIndex()) {
-                var text = "${region.name} ${region.cases}"
-
-                val extra = ArrayList<String>()
-                if (region.deaths != 0) extra.add("${region.deaths} dead")
-                if (region.recoveries != 0) extra.add("${region.recoveries} recovered")
-                if (extra.isNotEmpty()) text += " (" + extra.joinToString(", ") + ")"
-
-                if (region.name == "Germany" || region.name == "Total") text = BOLD + text + RESET
-                if (i != 0) text = ", $text"
-
-                if (text.length + messageBuilder.length > MAX_MESSAGE_LENGTH - 3) {
-                    messageBuilder.append("…")
-                    break
-                }
-                messageBuilder.append(text)
-            }
-            command.channel.sendMessageSafe(messageBuilder.toString())
+            command.channel.sendMessageSafe("nCov cases: Total 79545 (2627 dead, 25179 recovered), Mainland China[c] 77150 (2593 dead, 24920 recovered), South Korea 833 (8 dead, 22 recovered), International conveyance[d] 691 (3 dead, 1 recovered), Italy 224 (5 dead, 2 recovered), Japan 154 (1 dead, 23 recovered), Singapore 90 (53 recovered), Hong Kong 79 (2 dead, 19 recovered), Iran 64 (12 dead, 3 recovered), United States 35 (7 recovered), Thailand 35 (21 recovered)")
```

Comment:
I'm sure this line can be written a bit more compact! Remember, compact code is good code!
I suggest something like: command.channel.sendMessageSafe("nCov cases: Total 79545 (2627✝, 25179🎉), 🇨🇳 77150 (2593✝, 24920🎉), 🇰🇷 833 (8✝, 22🎉), 🧟boat 691 (3✝, 1🎉), 🇮🇹 224 (5✝, 2🎉), 🇯🇵 154 (1✝, 23🎉), 🇸🇬 90 (53🎉), 🇹🇼 79 (2✝, 19🎉), 🇮🇷 64 (12✝, 3🎉), 💵 35 (7🎉), 🇹🇭 35 (21🎉)")

Reply:
I'm sure there is some kind of StringBuilder Class, which could make the proposed line more readable.

Reply:
You mean a StringBuilderFactoryFactory?

Reply:
more like StringBuilderFactoryGeneratorFactory but yeah